src/basic/macro.h

```diff
@@ -247,12 +247,45 @@ static inline unsigned long ALIGN_POWER2(unsigned long u) {
                 (__x / __y + !!(__x % __y));                            \
         })
 
+#ifdef __COVERITY__
+
+/* Use special definitions of assertion macros in order to prevent
+ * false positives of ASSERT_SIDE_EFFECT on Coverity static analyzer
+ * for uses of assert_se() and assert_return().
+ *
+ * These definitions make expression go through a (trivial) function
+ * call to ensure they are not discarded. Also use ! or !! to ensure
+ * the boolean expressions are seen as such.
+ */
+
+extern void __coverity_panic__(void);
+
+static inline int __coverity_check__(int condition) {
```

Comment:
wouldn't bool be better for both types here, than int?

Reply:
In principle, yes. But there seems to be a lot of magic in Coverity, so I'm not sure whether using a bool here would make it warn... So I'm trying to stick as close as possible to the suggestion I got in that thread (even though I had to adapt it a bit to work on assert_return() as well.)